Problem compiling with MSYS2 UCRT64

<!--
IMPORTANT: Do NOT use GitHub to post any questions or support requests!
           They will be closed immediately and ignored.

Questions should go to the gdal-dev mailing list at
https://lists.osgeo.org/mailman/listinfo/gdal-dev
or other support forums. GitHub issues are for bug
reports and suggestions for new features.

If you think there is an issue with coordinate order in GDAL 3.0, then it is
likely an intended behavior. See https://github.com/OSGeo/gdal/issues/1974
for more explanations.

The GDAL project is made of contributions from various individuals and
organizations, each with their own focus. The issue you are facing is
not necessarily in the priority list of those contributors and consequently
there is no guarantee that it will be addressed in a timely manner.
If this bug report or feature request is high-priority for you, and you
cannot address it yourself, we suggest engaging a GDAL developer or support
organisation and financially sponsoring a fix.

-->

## Expected behavior and actual behavior.

I am trying to compile GDAL without plugins, depending only on PROJ, but the compilation fails with some errors.

## Steps to reproduce the problem.

download the source code, install MSYS2+UCRT64 build tools, run:

```
mkdir buid
cd build
cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DGDAL_BUILD_OPTIONAL_DRIVERS=OFF -DOGR_BUILD_OPTIONAL_DRIVERS=OFF -DGDAL_USE_EXTERNAL_LIBS=OFF ../
ninja
```

The output is:

```
[7/594] Building CXX object port/CMakeFiles/cpl.dir/cplgetsymbol.cpp.obj
FAILED: port/CMakeFiles/cpl.dir/cplgetsymbol.cpp.obj
C:\msys64\ucrt64\bin\c++.exe -DDONT_DEPRECATE_SPRINTF -DENABLE_DEFLATE64 -DGDAL_COMPILATION -DHAVE_AVX_AT_COMPILE_TIME -DHAVE_LIBZ -DHAVE_SSE_AT_COMPILE_TIME -DHAVE_SSSE3_AT_COMPILE_TIME -DINST_DATA="\"C:/Program Files (x86)/
gdal/share/gdal\"" -DRENAME_INTERNAL_ZLIB_SYMBOLS -DSYSCONFDIR="\"C:/Program Files (x86)/gdal/etc\"" -DZIP_SUPPORT -ID:/gdal-3.7.0/port -ID:/gdal-3.7.0/bb/port -ID:/gdal-3.7.0/ogr/ogrsf_frmts/geojson/libjson -ID:/gdal-3.7.0/f
rmts/zlib -ID:/gdal-3.7.0/frmts/zlib/contrib/infback9 -fvisibility=hidden -O3 -DNDEBUG -std=gnu++17 -Wall -Wextra -Winit-self -Wunused-parameter -Wmissing-declarations -Wlogical-op -Wshadow -Wmissing-include-dirs -Wformat -We
rror=format-security -Wno-format-nonliteral -Werror=vla -Wno-clobbered -Wdate-time -Wnull-dereference -Wextra-semi -Wfloat-conversion -Wnon-virtual-dtor -Woverloaded-virtual -Wsuggest-override -fno-operator-names -Wzero-as-nu
ll-pointer-constant -Wold-style-cast -Weffc++ -MD -MT port/CMakeFiles/cpl.dir/cplgetsymbol.cpp.obj -MF port\CMakeFiles\cpl.dir\cplgetsymbol.cpp.obj.d -o port/CMakeFiles/cpl.dir/cplgetsymbol.cpp.obj -c D:/gdal-3.7.0/port/cplge
tsymbol.cpp
In file included from C:/msys64/ucrt64/include/c++/13.1.0/stdlib.h:36,
                 from D:/gdal-3.7.0/port/cpl_port.h:140,
                 from D:/gdal-3.7.0/port/cplgetsymbol.cpp:30:
C:/msys64/ucrt64/include/c++/13.1.0/cstdlib:141:11: error: 'at_quick_exit' has not been declared in '::'
  141 |   using ::at_quick_exit;
      |           ^~~~~~~~~~~~~
C:/msys64/ucrt64/include/c++/13.1.0/cstdlib:164:11: error: 'quick_exit' has not been declared in '::'
  164 |   using ::quick_exit;
      |           ^~~~~~~~~~
C:/msys64/ucrt64/include/c++/13.1.0/stdlib.h:43:14: error: 'at_quick_exit' has not been declared in 'std'
   43 |   using std::at_quick_exit;
      |              ^~~~~~~~~~~~~
C:/msys64/ucrt64/include/c++/13.1.0/stdlib.h:46:14: error: 'quick_exit' has not been declared in 'std'
   46 |   using std::quick_exit;
      |              ^~~~~~~~~~
ninja: build stopped: subcommand failed.

```

## Operating system

Windows 10 Pro

## GDAL version and provenance

GDAL 3.6.x and 3,7

--- a/port/cpl_port.h
+++ b/port/cpl_port.h
@@ -114,15 +114,6 @@
 /*      MinGW stuff                                                     */
 /* ==================================================================== */
 
-/* We need __MSVCRT_VERSION__ >= 0x0700 to have "_aligned_malloc" */
-/* Latest versions of mingw32 define it, but with older ones, */
-/* we need to define it manually */
-#if defined(__MINGW32__)
-#ifndef __MSVCRT_VERSION__
-#define __MSVCRT_VERSION__ 0x0700
-#endif
-#endif
-
 /* Needed for std=c11 on Solaris to have strcasecmp() */
 #if defined(GDAL_COMPILATION) && defined(__sun__) &&                           \
     (__STDC_VERSION__ + 0) >= 201112L && (_XOPEN_SOURCE + 0) < 600

--- a/port/cpl_vsisimple.cpp
+++ b/port/cpl_vsisimple.cpp
@@ -34,6 +34,17 @@
  *
  ****************************************************************************/
 
+// Must be done before including cpl_port.h
+// We need __MSVCRT_VERSION__ >= 0x0700 to have "_aligned_malloc"
+// Latest versions of mingw32 define it, but with older ones,
+// we need to define it manually.
+#if defined(__MINGW32__)
+#include <windows.h>
+#ifndef __MSVCRT_VERSION__
+#define __MSVCRT_VERSION__ 0x0700
+#endif
+#endif
+
 #include "cpl_port.h"
 #include "cpl_vsi.h"
 

